packages/browser/src/core/plugin/index.ts

```diff
@@ -1,36 +1,12 @@
-import { Analytics } from '../analytics'
-import { Context } from '../context'
+import type { CorePlugin } from '@segment/analytics-core'
+import type { DestinationMiddlewareFunction } from '../../plugins/middleware'
+import type { Analytics } from '../analytics'
+import type { Context } from '../context'
 
-interface PluginConfig {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  options?: any
-  priority?: 'critical' | 'non-critical' // whether AJS should expect this plugin to be loaded before starting event delivery
-}
-
-// enrichment - modifies the event. Enrichment can happen in parallel, by reducing all changes in the final event. Failures in this stage could halt event delivery.
-// destination - runs in parallel at the end of the lifecycle. Cannot modify the event, can fail and not halt execution.
-// utility - do not affect lifecycle. Should be run and executed once. Their `track/identify` calls don't really do anything. example
-
-export interface Plugin {
-  name: string
-  version: string
-  type: 'before' | 'after' | 'destination' | 'enrichment' | 'utility'
-  alternativeNames?: string[]
+export interface Plugin extends CorePlugin<Context, Analytics> {}
 
-  isLoaded: () => boolean
-  load: (
-    ctx: Context,
-    instance: Analytics,
-    config?: PluginConfig
-  ) => Promise<unknown>
-
-  unload?: (ctx: Context, instance: Analytics) => Promise<unknown> | unknown
-
-  ready?: () => Promise<unknown>
-  track?: (ctx: Context) => Promise<Context> | Context
-  identify?: (ctx: Context) => Promise<Context> | Context
-  page?: (ctx: Context) => Promise<Context> | Context
-  group?: (ctx: Context) => Promise<Context> | Context
-  alias?: (ctx: Context) => Promise<Context> | Context
-  screen?: (ctx: Context) => Promise<Context> | Context
+export interface DestinationPlugin extends Plugin {
```

Comment:
Is this type only used internally?

Since users can write their own plugins, making `addMiddleware` a required method would be breaking (and not sure we want to expose this functionality directly.) If it's just internally used then no concerns as long as we aren't casting user-plugins to this type.

Reply:
This is only used internally